File: Desktop/Desktop.Data/Habitacion.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class Habitacion
    {
        private string _url;
        public int Id { get; set; }
        public int Numero { get; set; }
        public int Camas { get; set; }
        public int Capacidad { get; set; }
        public int Banos { get; set; }
        public bool Estado { get; set; }
        public int HotelId { get; set; }
        public Hotel Hotel { get; set; }

        public Habitacion()
        {
            _url = "http://127.0.0.1:3000/";
            GetHelper<Habitacion>.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
        }
        public bool CrearHabitacion(Habitacion habitacion)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "numero", habitacion.Numero.ToString() },
                { "capacidad", habitacion.Capacidad.ToString() },
                { "camas", habitacion.Camas.ToString() },
                { "banos", habitacion.Banos.ToString() },
                { "hotelId", habitacion.HotelId.ToString() },
            };
            return Conexion.Post(queryParams, "crearHabitacion");
        }

        public bool ActualizarHabitacion(Habitacion habitacion)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", habitacion.Id.ToString() },
                { "numero", habitacion.Numero.ToString() },
                { "capacidad", habitacion.Capacidad.ToString() },
                { "camas", habitacion.Camas.ToString() },
                { "banos", habitacion.Banos.ToString() },
                { "hotelId", habitacion.HotelId.ToString() },
            };
            return Conexion.Post(queryParams, "actualizarHabitacion");
        }

        public bool ActualizarEstadoHabitacion(int id, bool estado)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", id.ToString() },
                { "estado", estado ? "1" : "0" },
            };
            return Conexion.Post(queryParams, "actualizarEstadoHabitacion");
        }

        public List<Habitacion> ObtenerHabitaciones()
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var res = Conexion.Cliente.GetStringAsync(_url + "obtenerHabitaciones").Result;
            List<Habitacion> habitaciones = JsonConvert.DeserializeObject<List<Habitacion>>(res);
            return habitaciones;
        }

        public List<Habitacion> ObtenerHabitacionPorId(int id)
        {

            return ObtenerHabitaciones().Where(h => h.HotelId == id).ToList();
        }

    }
}

File: server/bussines/habitacion.js
const habitacionDao = require("../dao/habitacionDao.js");

const crearHabitacion = function(req, res){
    let habitacion = 
    {
        numero: req.body.numero,
        capacidad: parseInt(req.body.capacidad),
        camas:  parseInt(req.body.camas),
        banos:  parseInt(req.body.banos),
        hotelId: parseInt(req.body.hotelId)
    };
    habitacionDao.crearHabitacion(habitacion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarHabitacion = function(req, res){
    let habitacion = 
    {
        id: parseInt(req.body.id),
        numero: req.body.numero,
        capacidad: parseInt(req.body.capacidad),
        camas:  parseInt(req.body.camas),
        banos:  parseInt(req.body.banos),
        hotelId: parseInt(req.body.hotelId)
    };
    habitacionDao.actualizarHabitacion(habitacion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const actualizarEstadoHabitacion = function(req, res){
    let habitacion = 
    {
        id : parseInt(req.body.id),
        estado : req.body.estado
    };
    habitacionDao.actualizarEstadoHabitacion(habitacion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const obtenerHabitacionPorId = function(req, res){
    let habitacion = 
    {
        id : parseInt(req.body.id)
    };
    habitacionDao.obtenerHabitacionPorId(habitacion).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};
const obtenerHabitaciones = function(req, res){
    habitacionDao.obtenerHabitaciones().then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

module.exports.crearHabitacion = crearHabitacion;
module.exports.actualizarHabitacion = actualizarHabitacion;
module.exports.actualizarEstadoHabitacion = actualizarEstadoHabitacion;
module.exports.obtenerHabitacionPorId = obtenerHabitacionPorId;
module.exports.obtenerHabitaciones = obtenerHabitaciones;
